File: backend/src/RealEstate.Application/DTOs/Properties/PropertyDetailsDto.cs
namespace RealEstate.Application.DTOs.Properties;

public sealed record PropertyDetailsDto(
    string Id,
    string Title,
    string Description,
    string Address,
    string City,
    decimal Price,
    string Type,
    int Bedrooms,
    int Bathrooms,
    decimal Area,
    string Status,
    string? MainImageUrl,
    string BrokerId,
    DateTime CreatedAt
);

File: frontend/src/features/properties/api/dtos/property-details.dto.ts
import type { PropertyType} from "@/entities/properties/model/types";
import type { PropertyStatus } from "@/entities/properties/model/types";

export type PropertyDetailsDto = {
  id: number;
  title: string;
  description: string;
  address: string;
  city: string;
  price: number;
  type: PropertyType;
  status: PropertyStatus;
  bedrooms: number;
  bathrooms: number;
  area: number;
  mainImageUrl?: string | null;
  brokerId: string;
  createdAt?: string;
  updatedAt?: string;
  // TODO: add imagesUrls: string[] - both for front and back
}
